### ISSUE TITLE
[header] <ul> and <ol> must only directly contain <li>, <script> or <template> elements

### ISSUE THREAD

**Issue by sandbergja** [pulibrary/allsearch_frontend#296]
### Steps to recreate

1. Go to https://allsearch.princeton.edu/?q=giraffe
2. Run axe dev tools
3. Note that you get the error `<ul> and <ol> must only directly contain <li>, <script> or <template> elements`

### URL

https://allsearch.princeton.edu/?q=giraffe

### Issues

List element has direct children that are not allowed: a

### Severity

Serious

### WCAG Conformance level (A, AA, AAA)

Level A

### User impact

When content elements other than list items are contained within a set of list elements, screen readers cannot inform the listener that they are listening to items within the list.

See [deque's page about this rule](https://dequeuniversity.com/rules/axe/4.8/list)

### Recommended fix

The `<a>` should be inside an `<li>`

### Acceptance criteria
- [ ] I can successfully use this feature with Voice Over in Safari.
- [ ] axe dev tools does not complain about `<ul> and <ol> must only directly contain <li>, <script> or <template> elements`

**Comment by sandbergja** [comment]
@christinach and I came up with this plan:

- Start using the LuxMenuBar component
- Add a prop to the LuxMenuBar component to separate the carets
- Add a theme to the LuxMenuBar component to match what is on library.psb-prod

**Comment by maxkadel** [comment]
To facilitate reviewable PRs, and since the site isn't in full production / active review right now, we can break it up into smaller chunks with less-than-ideal styling.
